phoenix-core/src/main/java/org/apache/phoenix/mapreduce/index/IndexUpgradeTool.java

```diff
@@ -0,0 +1,547 @@
+                        .hasCoprocessor(GlobalIndexChecker.class.getName())) {
+                    if (!dryRun) {
+                        indexTableDescBuilder.addCoprocessor(GlobalIndexChecker.class.getName(),
+                                null, QueryServicesOptions.DEFAULT_COPROCESSOR_PRIORITY, prop);
+                    }
+                    LOGGER.info("Loaded GlobalIndexChecker coprocessor on index table " + indexName);
+                } else {
+                    LOGGER.info("GlobalIndexChecker coprocessor on index table " + indexName + " is already loaded");
+                }
+            } else {
+                if (admin.getDescriptor(TableName.valueOf(indexName))
+                        .hasCoprocessor(GlobalIndexChecker.class.getName())) {
+                    if (!dryRun) {
+                        indexTableDescBuilder.removeCoprocessor(GlobalIndexChecker.class.getName());
+                    }
+                    LOGGER.info("Unloaded GlobalIndexChecker coprocessor on index table "+indexName);
+                } else {
+                    LOGGER.info("GlobalIndexChecker coprocessor on index table " + indexName + " is already unloaded");
+                }
+            }
+            if (!dryRun) {
+                admin.modifyTable(indexTableDescBuilder.build());
+            }
+        }
+    }
+
+
+    private void rebuildIndexes(String dataTable, Configuration conf) {
+        String schema = SchemaUtil.getSchemaNameFromFullName(dataTable);
+        String table = SchemaUtil.getTableNameFromFullName(dataTable);
+        for(String index: rebuildMap.get(dataTable)) {
+            String outFile = "/tmp/index_rebuild_"+index+"_" + UUID.randomUUID().toString();
+            String indexName = SchemaUtil.getTableNameFromFullName(index);
+            String [] args = { "-s", schema, "-dt", table, "-it", indexName, "-direct", "-op", outFile};
+            indexingTool.setConf(conf);
+            if (syncRebuild) {
+                List<String>list = Arrays.asList(args);
+                list.add("-runfg");
+                args = list.toArray(new String[list.size()]);
+            }
+            try {
+                LOGGER.info("Rebuilding index "+ index);
+                if(!dryRun) {
+                    indexingTool.run(args);
+                }
+            } catch (Exception e) {
+                LOGGER.severe("Something went wrong while building the index "+ index +" "+e);
+                System.exit(-1);            }
+        }
+    }
+
+    private void extractTablesAndIndexes(PhoenixConnection conn) {
+        String [] tables = inputTables.trim().split(",");
+        PTable dataTable = null;
+        try {
+            for (String tableName : tables) {
+                HashSet<String> physicalIndexes = new HashSet<>();
+                dataTable = PhoenixRuntime.getTableNoCache(conn, tableName);
+                String physicalTableName = dataTable.getPhysicalName().getString();
```

Comment:
We should check if the table is a data table and if not we should skip and log and continue. For example, if an index table is included in the inputTables, this should not be silently ignored.